sql/src/test/java/org/apache/druid/sql/calcite/util/SqlTestFramework.java

```diff
@@ -550,16 +569,17 @@ public QueryLifecycleFactory queryLifecycleFactory(final Injector injector)
   private final AuthorizerMapper authorizerMapper = CalciteTests.TEST_AUTHORIZER_MAPPER;
   private final SqlEngine engine;
 
-  private SqlTestFramework(Builder builder)
+  private SqlTestFramework(Builder builder) throws Exception
   {
     this.builder = builder;
     this.componentSupplier = builder.componentSupplier;
+    builder.componentSupplier.beforeNewFrameworkCreateHook();
     Properties properties = new Properties();
     this.componentSupplier.gatherProperties(properties);
     Injector startupInjector = new StartupInjectorBuilder()
         .withProperties(properties)
         .build();
-    DruidInjectorBuilder injectorBuilder = new CoreInjectorBuilder(startupInjector)
+    CoreInjectorBuilder injectorBuilder = (CoreInjectorBuilder) new CoreInjectorBuilder(startupInjector)
```

Comment:
`ignoreLoadScopes` and `addModule` returns `DruidInjectorBuilder` ;
the cast was needed more-or-less because `ServiceInjector` wants a `CoreInjectorBuilder`

I'll submit a separate PR to fix the builders